kotlin/internal/jvm/jvm.bzl

```diff
@@ -161,6 +161,11 @@ _common_attr = utils.add_dicts(
             default = [],
             allow_files = False,
         ),
+        "friend": attr.label(
+            doc = """A single Kotlin dep which allows this code to access internal members of the given dependency.
+             Currently uses the output jar of the module -- i.e., exported deps won't be included.""",
```

Comment:
This example helped me understand what you were talking about: https://youtrack.jetbrains.com/issue/KT-34251

Reply:
Also: https://github.com/bazelbuild/rules_kotlin/issues/211